build.cake

```diff
@@ -0,0 +1,567 @@
+#addin "Cake.FileHelpers"
+#addin "Cake.Json"
+
+using System.Text.RegularExpressions;
+using System.ComponentModel;
+
+// Basic arguments
+var target = Argument("target", "Default");
+var configuration = Argument("configuration", "Release");
+// Optional arguments
+var testConfiguration = Argument("test-configuration", "Debug");
+var installFolder = Argument("install-path", IsRunningOnWindows() ?
+                        $"{EnvironmentVariable("USERPROFILE")}/.omnisharp/local" : "~/.omnisharp/local");
+
+// Working directory
+var workingDirectory = (new CakeEnvironment()).WorkingDirectory;
+
+// System specific shell configuration
+var shell = IsRunningOnWindows() ? "powershell" : "bash";
+var shellArgument = IsRunningOnWindows() ? "/Command" : "-C";
+var shellExtension = IsRunningOnWindows() ? "ps1" : "sh";
+
+/// <summary>
+///  Class representing build.json
+/// </summary>
+public class BuildPlan
+{
+    public IDictionary<string, string[]> TestProjects { get; set; }
+    public string BuildToolsFolder { get; set; }
+    public string ArtifactsFolder { get; set; }
+    public string DotNetFolder { get; set; }
+    public string DotNetInstallScriptURL { get; set; }
+    public string DotNetChannel { get; set; }
+    public string DotNetVersion { get; set; }
+    public string[] Frameworks { get; set; }
+    public string[] Rids { get; set; }
+    public string MainProject { get; set; }
+}
+var buildPlan = DeserializeJsonFromFile<BuildPlan>($"{workingDirectory}/build.json");
+
+// Folders and tools
+var dotnetFolder = $"{workingDirectory}/{buildPlan.DotNetFolder}";
+var dotnetcli = $"{dotnetFolder}/cli/dotnet";
+var toolsFolder = $"{workingDirectory}/tools";
+var xunitRunner = "xunit.runner.console";
+
+var sourceFolder = $"{workingDirectory}/src";
+var testFolder = $"{workingDirectory}/tests";
+
+var artifactFolder = $"{workingDirectory}/{buildPlan.ArtifactsFolder}";
+var publishFolder = $"{artifactFolder}/publish";
+var logFolder = $"{artifactFolder}/logs";
+var packageFolder = $"{artifactFolder}/package";
+
+/// <summary>
+///  Retrieve the default local RID from .NET CLI.
+/// </summary>
+/// <param name="dotnetcli">Full path to .NET CLI binary</param>
+/// <returns>Default RID</returns>
+string GetLocalRuntimeID(string dotnetcli)
```

Comment:
The implementation of this is not stable. We don't guarantee `dotnet --version` output's format. Use `PlatformServices` instead.